.github/workflows/gsprocessing_workflow.yml

```diff
@@ -0,0 +1,120 @@
+# Graphstorm CI for GSProcessing Only
+# Test execution order - [lint] -> [pytest]
+name: gsprocessing_workflow
+
+on:
+  push:
+    branches:
+      - main
+      - ci_dev
+    paths:
+      - '.github/workflow_scripts/gsprocessing.sh'
+      - 'graphstorm-processing/*'
+      - '.github/workflows/gsprocessing_workflow.yml'
+    workflow_dispatch:
+  pull_request_target:
+    paths:
+      - '.github/workflow_scripts/gsprocessing.sh'
+      - 'graphstorm-processing/*'
+      - '.github/workflows/gsprocessing_workflow.yml'
+
+permissions:
+  id-token: write
+  pull-requests: write
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+# All test scripts split in individual .sh files and moved to .github/workflow scripts
+env:
+  COMMAND-LINT: chmod +x ./lint_check.sh && ./lint_check.sh
+  COMMAND-GSProcessing: chmod +x ./gsprocessing.sh && ./gsprocessing.sh
+
+jobs:
+  lint_check:
+    if: ${{ github.event.label.name != 'draft' }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: arn:aws:iam::698571788627:role/github-oidc-role
+        aws-region: us-east-1
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      shell: bash
+      run: |
+        python3 -m pip install --upgrade pip
+        python3 -m pip install pytest
```

Comment:
I do wonder why we're installing pytest on the Github runner, since the tests will run on Batch. We can skip this right?